qa/rpc-tests/test_framework/util.py

```diff
@@ -377,6 +393,66 @@ def initialize_chain_clean(test_dir, num_nodes):
     for i in range(num_nodes):
         initialize_datadir(test_dir, i)
 
+def persistent_cache_path(cache_behavior):
+    return os.path.join(
+        os.path.dirname(os.path.dirname(os.path.realpath(__file__))), 
+        'cache', 
+        cache_behavior
+    )
+
+def persistent_cache_exists(cache_behavior):
+    cache_path = persistent_cache_path(cache_behavior)
+    return os.path.exists(cache_path)
+
+# Clean up, zip, and persist the generated datadirs. Record the generation
+# time so that we can correctly set the system clock offset in tests that
+# restore their node states using the resulting files.
+def persist_node_caches(tmpdir, cache_behavior, num_nodes):
+    cache_path = persistent_cache_path(cache_behavior)
+    if os.path.exists(cache_path):
```

Comment:
Not suggesting `isdir` here, because a file in the way should also cause an exception.